jupiter-tests/src/test/java/org/junit/jupiter/engine/extension/CloseablePathTests.java

```diff
@@ -95,72 +113,107 @@ void cleanupRoot() throws IOException {
 			delete(root);
 		}
 
-		@Test
 		@DisplayName("succeeds if the factory returns a directory")
-		void factoryReturnsDirectory() throws Exception {
-			TempDirFactory factory = spy(new Factory(createDirectory(root.resolve("directory"))));
+		@ParameterizedTest
+		@ElementTypeSource
+		void factoryReturnsDirectoryDynamic(Class<?> elementType) throws IOException {
+			TempDirFactory factory = (elementContext, extensionContext) -> createDirectory(root.resolve("directory"));
 
-			closeablePath = TempDirectory.createTempDir(factory, DEFAULT, elementContext, extensionContext);
+			closeablePath = TempDirectory.createTempDir(factory, DEFAULT, elementType, elementContext,
+				extensionContext);
 			assertThat(closeablePath.get()).isDirectory();
 
 			delete(closeablePath.get());
 		}
 
-		@Test
 		@DisplayName("succeeds if the factory returns a symbolic link to a directory")
-		@DisabledOnOs(OS.WINDOWS)
-		void factoryReturnsSymbolicLinkToDirectory() throws Exception {
+		@ParameterizedTest
+		@ElementTypeSource
+		@DisabledOnOs(WINDOWS)
+		void factoryReturnsSymbolicLinkToDirectory(Class<?> elementType) throws IOException {
 			Path directory = createDirectory(root.resolve("directory"));
-			TempDirFactory factory = spy(new Factory(createSymbolicLink(root.resolve("symbolicLink"), directory)));
+			TempDirFactory factory = (elementContext,
+					extensionContext) -> createSymbolicLink(root.resolve("symbolicLink"), directory);
 
-			closeablePath = TempDirectory.createTempDir(factory, DEFAULT, elementContext, extensionContext);
+			closeablePath = TempDirectory.createTempDir(factory, DEFAULT, elementType, elementContext,
+				extensionContext);
 			assertThat(closeablePath.get()).isDirectory();
 
 			delete(closeablePath.get());
 			delete(directory);
 		}
 
+		@DisplayName("succeeds if the factory returns a directory on a non-default file system for a Path annotated element")
 		@Test
+		void factoryReturnsDirectoryOnNonDefaultFileSystemWithPath() throws IOException {
+			TempDirFactory factory = spy(new JimfsFactory());
```

Comment:
:question: We're not using the spy in this test, are we?
```suggestion
			TempDirFactory factory = new JimfsFactory();
```

Reply:
Indeed, copy-paste leftover. Thanks for spotting it!